plugins/source/gcp/resources/services/securitycenter/findings_fetch.go

```diff
@@ -2,28 +2,62 @@ package securitycenter
 
 import (
 	"context"
+	"fmt"
+	"time"
 
 	"google.golang.org/api/iterator"
 
 	pb "cloud.google.com/go/securitycenter/apiv1/securitycenterpb"
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugins/source/gcp/client"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
 	securitycenter "cloud.google.com/go/securitycenter/apiv1"
 )
 
-func fetchFindings(parent string) func(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
-	return func(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
-		c := meta.(*client.Client)
-		req := &pb.ListFindingsRequest{
+const pageSize = 1000
+
+func getRequest(ctx context.Context, c *client.Client, table string, parent string) *pb.ListFindingsRequest {
+	filter, err := c.Backend.Get(ctx, table, c.ID())
+	if err != nil {
+		c.Logger().Warn().Str("table", table).Msgf("failed to get filter %s", err.Error())
+	}
+	var req *pb.ListFindingsRequest
+	if filter == "" || err != nil {
+		req = &pb.ListFindingsRequest{
 			Parent:   parent,
-			PageSize: 1000,
+			OrderBy:  "event_time",
+			PageSize: pageSize,
 		}
+	} else {
+		req = &pb.ListFindingsRequest{
+			Parent:   parent,
+			OrderBy:  "event_time",
+			PageSize: pageSize,
+			Filter:   filter,
+		}
+	}
+	return req
+}
+
+func setBackendState(ctx context.Context, c *client.Client, table string, lastEventTime *timestamppb.Timestamp) {
+	err := c.Backend.Set(ctx, table, c.ID(), fmt.Sprintf(`event_time >= "%s"`, lastEventTime.AsTime().Format(time.RFC3339)))
```

Comment:
I chose `>=` to have a bit of buffer. We could do `>` too but I think it's ok to have some overlap

Reply:
Storing the filter string itself (and not the int/timestamp) is a unique approach and might make it hard to "port" old states later on (if the filter format changes in a newer `pb`)

Reply:
It's actually intentional if someone wants to provide a different filter/state manually (was also a request).
I think if the filter format changes we can drop old states and do a full refresh